src/components/RecoilAppController/RecoilAppController.tsx

```diff
@@ -132,10 +132,10 @@ export const RecoilAppController = () => {
   }, [web3Context.library]);
 
   useEffect(() => {
-    if (selectedCircleId !== undefined) {
+    if (selectedCircleId !== undefined && myProfile) {
```

Comment:
is there any risk of recreating the race condition that you fixed in [35d5079]?(https://github.com/coordinape/coordinape/commit/35d5079555bcfec2eeec3f921d5f69d7a71da8bb#diff-e98035a9e3e99aed9d5cdae22c5662c49215d334c22be8439c240720dc93fb3cL123)

Reply:
This whole flow needs a rethink. There is the bug Regan found, which I think this fixed, but then I found an entirely different one..

I'd like to rewrite all of this logic so it is less prone to race conditions, and it intersects with the access control ticket w/ zashton, and exploring using never ending promises as initial value of some atoms. We could also store all the wallets people have used and make it easier to manage multiple wallets. And! people should be able to view other circles in their org!